src/resources/lib/abstract-resource.js

```diff
@@ -479,16 +486,31 @@ module.exports = class AbstractResource extends Restypie.Resources.AbstractCoreR
    * @return {Promise}
    */
   parseBody(bundle) {
+    const self = this;
     let supported = ['application/json', 'multipart/form-data'];
+    let parsed;
     switch (typeIs(bundle.req, supported)) {
       case 'application/json':
-        return this._parseJSON(bundle);
+        parsed = this._parseJSON(bundle);
+        break;
       case 'multipart/form-data':
-        return this._parseMultipart(bundle);
+        parsed = this._parseMultipart(bundle);
+        break;
+    }
+
+    if (!parsed) {
+      let headers = bundle.req.headers['content-type'];
+      return Promise.reject(new Restypie.TemplateErrors.UnsupportedFormat({ expected: supported, value: headers }));
     }
 
-    let headers = bundle.req.headers['content-type'];
-    return Promise.reject(new Restypie.TemplateErrors.UnsupportedFormat({ expected: supported, value: headers }));
+    return Promise.resolve(parsed)
```

Comment:
Oh ! `parsed` is a promise ? Maybe use a more descriptive naming, I got confused thinking it was an object. Also you can do `parserPromise.then...`